SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/StorageAdapter.cs

```diff
@@ -1,158 +1,193 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.AppCenter.Utils;
-using Microsoft.AppCenter.Utils.Files;
-using SQLite;
 using System;
 using System.Collections.Generic;
-using System.Linq.Expressions;
-using System.Reflection;
-using System.Threading.Tasks;
+using System.Linq;
+using SQLitePCL;
 
 namespace Microsoft.AppCenter.Storage
 {
     internal class StorageAdapter : IStorageAdapter
     {
-        private SQLiteAsyncConnection _dbConnection;
-        internal Directory _databaseDirectory;
-        private readonly string _databasePath;
+        private sqlite3 _db;
 
-        public StorageAdapter(string databasePath)
+        public void Initialize(string databasePath)
         {
-            _databasePath = databasePath;
-            var databaseDirectoryPath = System.IO.Path.GetDirectoryName(databasePath);
-            if (databaseDirectoryPath != string.Empty)
+            int result;
+            try
+            {
+                raw.SetProvider(new SQLite3Provider_e_sqlite3());
+                result = raw.sqlite3_open(databasePath, out _db);
+            }
+            catch (Exception e)
             {
-                _databaseDirectory = new Directory(databaseDirectoryPath);
+                throw new StorageException("Failed to open database connection.", e);
+            }
+            if (result != raw.SQLITE_OK)
+            {
+                var errorMessage = raw.sqlite3_errmsg(_db);
+                throw new StorageException($"Failed to open database connection, result={result}\n\t{errorMessage}");
             }
         }
 
-        public async Task CreateTableAsync<T>() where T : new()
+        public void Dispose()
         {
-            try
+            if (_db == null)
             {
-                // In SQLite-net 1.5 return type was changed.
-                // Using reflection to accept newer library version.
-                var task = (Task)_dbConnection.GetType()
-                    .GetMethod("CreateTableAsync", new[] { typeof(CreateFlags) })
-                    .MakeGenericMethod(typeof(T))
-                    .Invoke(_dbConnection, new object[] { CreateFlags.None });
-                await task.ConfigureAwait(false);
-            }
-            catch (SQLiteException e)
-            {
-                throw ToStorageException(e);
+                AppCenterLog.Error(AppCenterLog.LogTag, "Trying to dispose storage adapter while database is null.");
```

Comment:
Per https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#dispose-and-disposeboolean,

> To help ensure that resources are always cleaned up appropriately, a Dispose method should be callable multiple times without throwing an exception.

Dispose pattern is supposed to be callable multiple times without side effects. We shouldn't log an error message if called a second time.